python/pyarrow/tests/test_compute.py

```diff
@@ -2985,6 +2985,17 @@ def test_list_slice_output_variable(start, stop, value_type, list_type):
     assert pylist == expected
 
 
+@pytest.mark.parametrize("fixed_size_list", (True, False))
+@pytest.mark.parametrize("type", (
+    lambda: pa.list_(pa.struct([('col', pa.int8())])),
```

Comment:
```suggestion
    lambda: pa.list_(pa.field(('col', pa.int8())),
```

The idea here is to change the field name of the _list_ child field (i.e. its values), not to have a list of structs.

Compare with the default constructor:
```
In [57]: pa.list_(pa.field("elements", pa.int64()))
Out[57]: ListType(list<elements: int64>)

In [58]: pa.list_(pa.int64())
Out[58]: ListType(list<item: int64>)
```

Reply:
Ah, now it's very clear. :+1: